examples/main.go

```diff
@@ -2,161 +2,297 @@ package main
 
 import (
 	"crypto/ecdsa"
-	"crypto/ed25519"
 	"crypto/elliptic"
-	"crypto/rand"
-	"crypto/sha256"
 	"encoding/base64"
-	"encoding/hex"
 	"encoding/json"
 	"fmt"
+	"io"
+	"math/rand"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
 	"time"
 
 	"github.com/go-jose/go-jose/v4"
-	"github.com/go-jose/go-jose/v4/jwt"
 )
 
+var (
+	staticRand io.Reader
+	now        = time.Date(2025, time.July, 7, 12, 0, 0, 0, time.UTC)
+)
+
+func init() {
+	// Generate deterministic examples. DEMO ONLY. DO NOT USE.
+	staticRand = &deterministicRandomReader{Reader: rand.New(rand.NewSource(42))}
+	jose.RandReader = staticRand
```

Comment:
this is nice for the signature values but overkill IMO for key generation and random string values